wallet-sdk/src/main/kotlin/org/stellar/walletsdk/Auth.kt

```diff
@@ -13,26 +13,20 @@ import org.stellar.walletsdk.util.OkHttpUtils
  * Authenticate to an external server using
  * [SEP-10](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md).
  *
- * @property accountAddress Stellar address of the account authenticating
  * @property webAuthEndpoint Authentication endpoint URL
  * @property homeDomain Domain hosting stellar.toml (
  * [SEP-1](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0001.md)) file
  * containing `WEB_AUTH_ENDPOINT` URL and `SIGNING_KEY`
- * @property memoId optional memo ID to distinguish the account
- * @property clientDomain optional domain hosting stellar.toml file containing `SIGNING_KEY`
- * @property networkPassPhrase Stellar network passphrase
+ * @property network Stellar network
  * @property walletSigner interface to define wallet client and domain (if using `clientDomain`)
  * signing methods
  * @property httpClient optional custom HTTP client, uses [OkHttpClient] by default
  */
 class Auth(
-  private val accountAddress: String,
   private val webAuthEndpoint: String,
   private val homeDomain: String,
-  private val memoId: String? = null,
-  private val clientDomain: String? = null,
-  private val networkPassPhrase: String = Network.TESTNET.toString(),
   private val walletSigner: WalletSigner,
+  private val network: Network = Network.TESTNET,
   private val httpClient: OkHttpClient = OkHttpClient()
 ) {
```

Comment:
What is the benefit of splitting these params between the class and inner methods? Is it to make the params list shorter?

To make the class more generic, we could move `walletSigner` to `authenticate` method.

Reply:
Yes, one of them is to make a list shorter.
But in general, the idea is that `Auth` object should be re-used. For example, I see that the flow should be
1. Create `Auth` class (one per server -> already pre-defined endpoints, etc)
2. Call `authenticate()` function for every user (i.e. once per account)

In general, class params -> generic configuration (per whole app lifecycle, per app, per server, per user, etc)
Method params -> less generic (one level down)

I didn't move `walletSigner`, because I think it should be static per server (unless `signWithDomainAccount` is called? In that case you are right, it can be moved to function parameters)